### PR TITLE
kconfig: logging: Add LOG_MIPI_SYST_ARGS_BUFFER_SIZE option.

### LINKED ISSUE #10: argBuf is too large to put on stack
the argBuf (on mipi_syst_api.c) was defined on stack and defaultly 1K, it means every call task stack should reserve 1K for this at least. otherwise it will be easily overflow of task.
is it possible to make it out of stack, like on bss?

### PR DESCRIPTION
**Pull request by aasthagr**
Add user configurable LOG_MIPI_SYST_ARGS_BUFFER_SIZE Kconfig
option to use it for macro MIPI_SYST_PCFG_PRINTF_ARGBUF_SIZE.
Moving macro definition of MIPI_SYST_PCFG_PRINTF_ARGBUF_SIZE
from mipi-sys-t library to platform.h which defaulted to 1kb
and resulted in stack overflow in some applications.
LOG_MIPI_SYST_ARGS_BUFFER_SIZE gives flexibilty to users when
maximum log length is known.

Fixes zephyrproject-rtos/mipi-sys-t#10

Signed-off-by: Aastha Grover <aastha.grover@intel.com>

### CODE REVIEW

File: subsys/logging/Kconfig.formatting [hunk 1]
```diff
@@ -52,6 +52,16 @@ config LOG_MIPI_SYST_CATALOG_ARGS_BUFFER_SIZE
 	  argument list needed for the MIPI Sys-T library for processing
 	  catalog messages.
 
+config LOG_MIPI_SYST_ARGS_BUFFER_SIZE
```
Comment (nordic-krch) [inline subsys/logging/Kconfig.formatting:55]: will it be used anywhere?

Comment (nordic-krch) [reply subsys/logging/Kconfig.formatting:55]: ok, i see that it is used by mipi repo

**Review by nordic-krch — DISMISSED** [review]

**Comment by nordic-krch** [comment]
Maybe add west manifest update to this PR?

**Comment by aasthagr** [comment]
> Maybe add west manifest update to this PR?

Added. Although will need to change the revision once the PR [#11](https://github.com/zephyrproject-rtos/mipi-sys-t/pull/11) will be merged.

**Review by MaureenHelm — DISMISSED** [review]

**Review by dcpleung — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**